src/main/scala/com/databricks/sparkdl/ImageUtils.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2017 Databricks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.databricks.sparkdl
+
+import java.awt.image.BufferedImage
+import java.awt.{Color, Image}
+
+import org.apache.spark.image.ImageSchema
+import org.apache.spark.sql.Row
+
+object ImageUtils {
+
+  /**
+   * Takes a Row image (spImage) and returns a Java BufferedImage. Currently supports 1 & 3
+   * channel images. If the image has 3 channels, we assume the channels are in BGR order.
+   *
+   * @param rowImage Image in spark.ml.image format.
+   * @return Java BGR BufferedImage.
+   */
+  private[sparkdl] def spImageToBufferedImage(rowImage: Row): BufferedImage = {
+    val height = ImageSchema.getHeight(rowImage)
+    val width = ImageSchema.getWidth(rowImage)
+    val channels = ImageSchema.getNChannels(rowImage)
+    val imageData = ImageSchema.getData(rowImage)
+    require(
+      imageData.length == height * width * channels,
+      s"""| Only one byte per channel is currently supported, got ${imageData.length} bytes for
+          | image of size ($height, $width, $channels).
+       """.stripMargin
+    )
+    val image = new BufferedImage(width, height, BufferedImage.TYPE_3BYTE_BGR)
+
+    var offset, h = 0
+    var r, g, b: Byte = 0
+    while (h < height) {
+      var w = 0
+      while (w < width) {
```

Comment:
(Migrated from the comments on https://github.com/databricks/spark-deep-learning/pull/62)
Shall we refactor this code into two loops, one for `channels == 1` and one for `channels == 3`. 
This way we only have to deal with the condition once outside the loop, making the code more performant.

Reply:
I think the jvm should handle the loop unswitching here. My understanding is that loop unswitching only has marginal benefits even in non-compiled languages, http://labs.septeni-technology.jp/none/loop-unswitching-a-performance-appraisal/.

My suggestion is that we focus on readability in this PR and push performance improvements to a separate PR which can include benchmarks.

Reply:
I agree that loop optimizations (e.g. unswitching), might not always provide the best performance. 
However, in this case, the [JVM loop unswitching bug](https://github.com/rholder/jvm-loop-unswitching-bug) might have a chilling effect on JVM to apply loop unswitching automatically.

Reply:
The argument in the [cited article](http://labs.septeni-technology.jp/none/loop-unswitching-a-performance-appraisal/) is flawed as the micro-benchmark is not correctly conducted.

Reply:
I agree that readability is important. In this case though, splitting it into two loops does not hinder readability as we don't have to deal with BGR conversion for mono channel images. 
I will leave the readability discussion to @sueann or @thunterdb.

Reply:
I will be fantastic if we can get another PR to focus on performance benchmarking.
Data ingestion is on the critical path for massively distributed deep learning workload.

Reply:
We can worry about performance once we have some users... :-)